### ISSUE TITLE
Add support for using proxy when downloading chapters

### ISSUE THREAD

**Issue by galpt** [KotatsuApp/kotatsu-dl#2]
The proxy feature (0ms.dev) is already implemented to the Kotatsu mobile app as an image optimization proxy (https://github.com/KotatsuApp/Kotatsu/issues/771).

Some benefits worth mentioning from using the proxy:
### Reducing server load from these manga websites
Actually, the proxy can be used to reduce burden to a manga source when multiple users are trying to download a chapter, so multiple requests will go to the proxy first and the proxy will check if that chapter is already in the cache or not. If already in the cache then there's no need to forward the request to the manga source, so it will reduce the server load from that specific source.

### Protecting user's IP address when downloading images
The proxy will make requests on behalf of the users. So that means this could prevent the user from getting blocked when downloading the images since the manga website will see the proxy IP addresses that are downloading the image links.

### Acts as a caching proxy for everyone
The proxy also acts as a global CDN that will probably always be faster compared to downloading from the manga websites directly without proxy. It's because home ISPs won't always give the best peering to cloud servers compared to a CDN that usually already has a pretty good peering to most ISPs and cloud providers.
So `User <-> Proxy as CDN <-> Image links` is usually faster than `User <-> Image links`.

**Comment by Koitharu** [comment]
Yes but a bit later, I guess this feature is not mandatory for downloading

**Comment by MatinSenPai** [comment]
for us, users in countries in Iran it's really a necessary feature

**Comment by ffact** [comment]
Guys, you can use nsproxy to redirect program traffic to any proxy.
